### PR TITLE
bpo-40744: Drop support for SQLite pre 3.7.3

### PR DESCRIPTION
**Pull request by erlend-aasland**
Remove code required to to support SQLite pre 3.7.3.

Integrate setup.py, docs and unit test changes from PR #17413

Style changes not related to [bpo-40744](https://bugs.python.org/issue40744) were dropped from the merge.
Berker Peksag's suggested documentation changes was included.

See https://github.com/python/cpython/pull/17413

Co-written-by: Berker Peksag <berker.peksag@gmail.com>
Co-written-by: Sergey Fedoseev <fedoseev.sergey@gmail.com>

https://bugs.python.org/issue40744

### CODE REVIEW

**Comment by erlend-aasland** [comment]
@pablogsal, would you mind if I request your review on this as well?

**Comment by pablogsal** [comment]
> @pablogsal, would you mind if I request your review on this as well?

No problem. Will try to give it a go soon. Feel free to ping me in a couple of days if I have not made a review by then :)

File: Misc/NEWS.d/next/Library/2020-05-30-08-10-23.bpo-40744.jKURVV.rst
Comment (pablogsal) [inline Misc/NEWS.d/next/Library/2020-05-30-08-10-23.bpo-40744.jKURVV.rst]: Could you add some extra explanation on *why* we are dropping support? (Basically some similar text as the one from the bpo)

Comment (erlend-aasland) [reply Misc/NEWS.d/next/Library/2020-05-30-08-10-23.bpo-40744.jKURVV.rst]: See commit 5d180f8. Maybe too much?

Comment (pablogsal) [reply Misc/NEWS.d/next/Library/2020-05-30-08-10-23.bpo-40744.jKURVV.rst]: Sorry, I should have been more explicit. I meant to add something referring to the fact that the module already cannot be built with sqlite3 pre 3.7.3. (So this is not a new restriction, just making it official). 

Don't worry too much, I can tweak this myself before landing

Comment (erlend-aasland) [reply Misc/NEWS.d/next/Library/2020-05-30-08-10-23.bpo-40744.jKURVV.rst]: All right, thank you!

File: Modules/_sqlite/module.c
Comment (pablogsal) [inline Modules/_sqlite/module.c:348]: Is possible to reach this point? If the shared object that loads does not have some of the functions that we are using it will be a link error, no?

Comment (erlend-aasland) [reply Modules/_sqlite/module.c:348]: Yes, I'm pretty sure you're right about that, so this check seems superfluous.

Comment (erlend-aasland) [reply Modules/_sqlite/module.c:348]: However, if the reason for requiring a specific version was, for example, a new flag, we'd need the runtime check. That's not the case now, so no need to keep the check. (Though it may be easier to remember to update the runtime check if it's there already.)

Comment (erlend-aasland) [reply Modules/_sqlite/module.c:348]: Let me know if you want it removed.

Comment (pablogsal) [reply Modules/_sqlite/module.c:348]: Is fine, we can leave it in case something goes terribly wrong or if some distribution packages stubs for the symbols (not sure why anybody would do that but I have seen it in the past).

**Review by pablogsal — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the review, @pablogsal !

**Comment by pablogsal** [comment]
Thanks, @erlend-aasland for the PR :)

**Comment by erlend-aasland** [comment]
Should this be backported to 3.9, or is it too late?

**Comment by pablogsal** [comment]
> Should this be backported to 3.9, or is it too late?

Too late IMHO, check my message on bpo

**Pull request merged**
